pkg/gateway/utils_test.go

```diff
@@ -88,22 +143,61 @@ func Test_getAttachedPolicy(t *testing.T) {
 				},
 				policy: &anv1alpha1.TargetGroupPolicy{},
 			},
-			expectedK8sClientReturnedPolicy: targetGroupPolicyHappyPath,
-			want:                            targetGroupPolicyHappyPath,
-			expectPolicyNotFound:            false,
+			expectedK8sClientReturnedPolicies: []core.Policy{targetGroupPolicyHappyPath},
+			want:                              []core.Policy{targetGroupPolicyHappyPath},
+			expectPolicyNotFound:              false,
+		},
+		{
+			name: "Get gateway attached IAMAuthPolicy, happy path",
+			args: args{
+				refObjNamespacedName: types.NamespacedName{
+					Namespace: ns,
+					Name:      "gw-1",
+				},
+				policy: &anv1alpha1.IAMAuthPolicy{},
+			},
+			expectedK8sClientReturnedPolicies: []core.Policy{iamAuthPolicyHappyPath},
+			want:                              []core.Policy{iamAuthPolicyHappyPath},
+			expectPolicyNotFound:              false,
+		},
+		{
+			name: "Get GRPCRoute attached AccessLogPolicy, happy path, be able to get all targetRef Matched policies",
+			args: args{
+				refObjNamespacedName: types.NamespacedName{
+					Namespace: ns,
+					Name:      "grpcroute-1",
+				},
+				policy: &anv1alpha1.AccessLogPolicy{},
+			},
+			expectedK8sClientReturnedPolicies: []core.Policy{accessLogPolicy1HappyPath, accessLogPolicy2HappyPath},
+			want:                              []core.Policy{accessLogPolicy1HappyPath, accessLogPolicy2HappyPath},
```

Comment:
`GetAttachedPolicies()` is able to return both 2 attatched accessLogPolicies

Reply:
What about 3?